lib/logstash/config/config_ast.rb

```diff
@@ -275,9 +277,17 @@ def compile
     end
   end
   class RValue < Node; end
-  class Value < RValue; end
+  class Value < RValue;
+    def text_value_utf8
+      Unicode.convert(text_value)
+    end
+  end
 
   module Unicode
+    def self.convert(text)
+      text.force_encoding(Encoding::UTF_8)
```

Comment:
good catch and I don't test anythings with unicode.
I'll add test and check if we need to transcode or just forcing the type is fine.